### PR TITLE
feat: support `JSOX` **stringified** `paramTypes` option

### LINKED ISSUE #233: [prettier-plugin-sql] Invalid paramTypes value. Expected , but received { named: [":"] }.
Hello and thanks for this awesome plugin.

I am trying to configure the `paramTypes` option as defined in [sql-formatter](https://github.com/sql-formatter-org/sql-formatter/blob/master/docs/paramTypes.md), but when I do:
```json
{
  "prettier": {
    "paramTypes": {
      "named": [
        ":"
      ]
    }
  }
}
```

I get the following error when running prettier:
```
[error] Invalid paramTypes value. Expected , but received { named: [":"] }.
```

I am not sure why or how to fix it.

### LINKED ISSUE #277: Unable to format SQL: Error: Parse error: Unexpected "{env}_s3_d"
Unable to format SQL: Error: Parse error: Unexpected "{env}_s3_datalake.{".
This is for parameterized SQL to query multiple tables like: 
 `SELECT 
     *,
     current_date - interval {interval} {period}
  FROM {database_name}.{table_name}`

It seems to fail due to the use of  `'{' , '}'`
Using ``` ' ` ' ``` doesn't help due to some dialects not supporting it
Is it possible to add char ignore or add a custom sql dialect for this sort of query?

### LINKED ISSUE #279: [prettier-plugin-sql] Does this support PostgreSQL meta-commands?
It seems to produce an error when using PostgreSQL commands like `\c` or `\connect`.

`test-docker/test-db/init/01_init.sql`
```sql
-- create db
CREATE DATABASE test_db;

-- connect test_db
\c test_db

-- create schema
CREATE SCHEMA test;
```

traceback:
```
test-docker/test-db/init/01_init.sql
[error] test-docker/test-db/init/01_init.sql: Error: Parse error: Unexpected "\c test" at line 5 column 1
[error]     at TokenizerEngine.createParseError (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/lexer/TokenizerEngine.js:53:12)
[error]     at TokenizerEngine.tokenize (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/lexer/TokenizerEngine.js:35:22)
[error]     at Tokenizer.tokenize (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/lexer/Tokenizer.js:16:47)
[error]     at LexerAdapter.tokenize (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/parser/createParser.js:16:76)
[error]     at LexerAdapter.reset (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/parser/LexerAdapter.js:17:24)
[error]     at Parser.feed (/Users/takecchi/PhpstormProjects/test-db/node_modules/nearley/lib/nearley.js:281:15)
[error]     at Object.parse (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/parser/createParser.js:26:18)
[error]     at Formatter.parse (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/formatter/Formatter.js:32:49)
[error]     at Formatter.format (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/formatter/Formatter.js:25:22)
[error]     at formatDialect (file:///Users/takecchi/PhpstormProjects/test-db/node_modules/sql-formatter/lib/sqlFormatter.js:77:57)
```

Do you have any workaround or plans to support it?

### PR DESCRIPTION
**Pull request by JounQin**
close #233
close #277
close #279

### CODE REVIEW

File: packages/sql/test/__snapshots__/fixtures.spec.ts.snap [hunk 1]
```diff
@@ -8,6 +8,189 @@ WITH
...
 
+exports[`parser and printer > should format all fixtures > 233.sql 1`] = `
+"CREATE TABLE
+  \\"test\\" (\\"id\\" uuid NOT NULL)
+WITH
+  (oids = false);
+"
+`;
+
+exports[`parser and printer > should format all fixtures > 277.sql 1`] = `
```
Comment (JounQin) [inline packages/sql/test/__snapshots__/fixtures.spec.ts.snap:19]: @nene Is this expected? I'm following https://github.com/sql-formatter-org/sql-formatter#my-sql-contains-templating-syntax-which-sql-formatter-fails-to-parse at https://github.com/un-ts/prettier/pull/301/files#diff-412bd64af2a1add9795f0f4806f1db87c422f799e7dc4bb2a83babafe0f65ab8R19-R24, but there is still an error on formatting. (We're still using `sql-formatter@^12.2.4` for now.

Comment (nene) [reply packages/sql/test/__snapshots__/fixtures.spec.ts.snap:19]: Yeah, the parameters currently cannot occur inside the `schema.table.column` syntax. As, well, it's not really a place where a prepared-statement parameter could occur. There are many other places where these parameters can't occur. Like inside strings. Or if you'd like to parameterize some SQL keyword - then really all bets are off.

But this specific limitation is easy to lift. Made a quick fix to make it work: https://github.com/sql-formatter-org/sql-formatter/commit/0875fe57867fcea1a6b421ccd0d5022d2ee9cab7

Comment (JounQin) [reply packages/sql/test/__snapshots__/fixtures.spec.ts.snap:19]: Thanks for you confirmation. I'll upgrade `sql-fomatter` in a seperate PR.

Comment (JounQin) [reply packages/sql/test/__snapshots__/fixtures.spec.ts.snap:19]: @nene Hi, when will your last fix commit been released?

Comment (nene) [reply packages/sql/test/__snapshots__/fixtures.spec.ts.snap:19]: This is now released as 13.1.0

**Pull request merged**